include/aws/http/private/hpack.h

```diff
@@ -35,16 +35,17 @@ void aws_hpack_static_table_clean_up(void);
 /* General HPACK API */
 AWS_HTTP_API
 struct aws_hpack_context *aws_hpack_context_new(struct aws_allocator *allocator, size_t max_dynamic_elements);
-
 AWS_HTTP_API
 void aws_hpack_context_destroy(struct aws_hpack_context *context);
 
 AWS_HTTP_API
-struct aws_http_header *aws_hpack_get_header(struct aws_hpack_context *context, size_t index);
-
+const struct aws_http_header *aws_hpack_get_header(struct aws_hpack_context *context, uint64_t index);
 AWS_HTTP_API
-int aws_hpack_find_index(struct aws_hpack_context *context, const struct aws_http_header *header, size_t *index);
-
+int aws_hpack_find_index(
+    struct aws_hpack_context *context,
+    const struct aws_http_header *header,
+    uint64_t *index,
+    bool *found_value);
```

Comment:
a little bit weird that the function has an int return type and also a `found_value` out param

does the int return type indicate catastrophic failure within the datastruture?

Reply:
`found_value` lets you know whether it found a value, or just a key associated with the header